osu.Game/Screens/Select/BeatmapDetails.cs

```diff
@@ -451,5 +434,65 @@ private void load(OsuColour colour)
                 valueText.Colour = colour.GrayB;
             }
         }
+
+        private class MetadataSegment : Container
+        {
+            private readonly OsuSpriteText header;
+            private readonly FillFlowContainer<OsuSpriteText> content;
+
+            private const int fade_time = 250;
+
+            public string HeaderText
+            {
+                set
+                {
+                    header.Text = value;
+                }
+            }
+
+            public string ContentText
+            {
+                set
+                {
+                    if (value == "")
+                        FadeOut(fade_time);
+                    else
+                    {
+                        FadeIn(fade_time);
+                        content.Children = value.Split(' ').Select(text => new OsuSpriteText

```

Comment:
in case you didn't get the message: remove this for now and make tags a special case (you should make a `Tag` class)